### PR TITLE
Make procdump abort on std::bad_cast; kill SUT if it occurs; Make procdump optional.

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

File: js/client/modules/@arangodb/crash-utils.js [hunk 1]
```diff
@@ -167,12 +171,38 @@ function analyzeCoreDumpWindows (instanceInfo) {
...
+          pu.serverCrashed = true;
+          arangod['exitStatus'] = {};
+          analyzeCrash(binary, arangod, options, "the process monitor commanded error");
+          Object.assign(arangod.exitStatus,
+                        killExternal(arangod.pid, abortSignal));
+          return false;
+        }
+      }
+    }
+    else return arangod.monitor.exitStatus;
```
[the remark this answers is not in the document]
Comment (dothebart) [reply js/client/modules/@arangodb/crash-utils.js]: this is inside of ` if (arangod.hasOwnProperty('monitor') )` - so it should be defined.

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1412/

File: js/client/modules/@arangodb/testing.js [hunk 2]
```diff
@@ -86,6 +86,8 @@ let optionsDocumentation = [
   '',
   '   - `disableMonitor`: if set to true on windows, procdump will not be attached.',
   '   - `rr`: if set to true arangod instances are run with rr',
+  '   - `exceptionFilter`: on windows you can use this to abort tests on specific exceptions',
+  '                        i.e. `bad_cast` to abort on throwing of std::bad_cast',
```
Comment (KVS85) [inline js/client/modules/@arangodb/testing.js]: Please, add example here on how to set multiple exceptions: i.e. are they coma-separated and should be quoted?

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1477/

**Pull request merged**